src/track/track.cpp

```diff
@@ -286,12 +285,13 @@ double Track::setBpm(double bpmValue) {
         // No beat grid available -> create and initialize
         double cue = getCuePoint().getPosition();
         mixxx::BeatsPointer pBeats(BeatFactory::makeBeatGrid(*this, bpmValue, cue));
-        setBeatsMarkDirtyAndUnlock(&lock, pBeats);
+        setBeatsMarkDirtyAndUnlock(&lock, pBeats, false);
```

Comment:
Shouldn't we keep the current bpm lock instead of resetting it unconditionally?

Reply:
false means that. I will add an enum to clarify this.